### ISSUE TITLE
C# HttpClient throws exception when trying to add signature

### ISSUE THREAD

**Issue by BerserkerDotNet** [kamsar/MicroCHAP#1]
C# HttpClient throws an FormatException when I'm trying to add auth header like this:

``` CSharp
client.Headers.Add("Authorization", _responseService.CreateSignature(challenge, url, additionalFactors));
```

It does so because according to [RFC-2617](http://www.ietf.org/rfc/rfc2617.txt) auth header has two parts: a scheme part, and a parameter part. Like this: 

```
Authorization: Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ==
```

I was able to workaround this by using TryAddWithoutValidation method, but please consider adding schema to auth header.

**Comment by kamsar** [comment]
Interesting, I guess I'd never run into this because I was using `WebClient` instead of `HttpClient`

It shouldn't be too hard to make it add a scheme - say `uCHAP` - and be able to accept schemed or schemeless headers for backwards compatibility.

**Comment by kamsar** [comment]
Fixed in 1.2.0, by altering the default HTTP header names (with a legacy compatible fallback) to `X-MC-MAC` and `X-MC-Nonce` to avoid entanglements with RFCs.

The client in 1.2.0 automatically uses the new header names. 1.1.x compatible clients should still work, though they are deprecated.
